core/controllers/platform_feature_test.py

```diff
@@ -0,0 +1,113 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for platform feature handler."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from constants import constants
+from core.domain import platform_feature_services as feature_services
+from core.domain import platform_parameter_domain as param_domain
+from core.domain import platform_parameter_registry as registry
+from core.platform import models
+from core.tests import test_utils
+
+memcache_services = models.Registry.import_memcache_services()
+
+
+class PlatformFeatureHandlerTest(test_utils.GenericTestBase):
+    """Tests for the PlatformFeatureHandler."""
+
+    def setUp(self):
+        super(PlatformFeatureHandlerTest, self).setUp()
+
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.user_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+        self.original_registry = registry.Registry.parameter_registry
+        self.original_feature_list = feature_services.ALL_FEATURES_LIST
+        self.original_feature_name_set = feature_services.ALL_FEATURES_NAMES_SET
+
+        param_names = ['parameter_a', 'parameter_b']
+        self.memcache_keys = [
+            param_domain.PlatformParameter.get_memcache_key(name)
+            for name in param_names]
+        memcache_services.delete_multi(self.memcache_keys)
+
+        registry.Registry.parameter_registry.clear()
+        self.param_1 = registry.Registry.create_platform_parameter(
+            'parameter_a', 'parameter for test', 'bool',
+            is_feature=True, feature_stage=param_domain.FEATURE_STAGES.dev)
+        self.param_2 = registry.Registry.create_platform_parameter(
+            'parameter_b', 'parameter for test', 'bool',
+            is_feature=True, feature_stage=param_domain.FEATURE_STAGES.prod)
+        registry.Registry.update_platform_parameter(
+            self.param_2.name, self.user_id, 'edit rules',
+            [
+                {
+                    'filters': [
+                        {
+                            'type': 'server_mode',
+                            'conditions': [['=', param_domain.SERVER_MODES.dev]]
+                        }
+                    ],
+                    'value_when_matched': True
+                }
+            ]
+        )
+
+        feature_services.ALL_FEATURES_LIST = param_names
+        feature_services.ALL_FEATURES_NAMES_SET = set(param_names)
+
+    def tearDown(self):
+        super(PlatformFeatureHandlerTest, self).tearDown()
+
+        feature_services.ALL_FEATURES_LIST = self.original_feature_list
+        feature_services.ALL_FEATURES_NAMES_SET = self.original_feature_name_set
+        registry.Registry.parameter_registry = self.original_registry
+
+    def test_post_handler_returns_correct_feature_flag_values(self):
```

Comment:
```suggestion
    def test_post_dev_mode_android_client_returns_correct_flag_values(self):
```

As discussed in earlier PRs, test names should be clearly describing the conditions being tested.

Reply:
Done.